src/test/java/org/broadinstitute/consent/http/cloudstore/GCSServiceTest.java

```diff
@@ -139,4 +140,30 @@ void testDeleteDocument() {
     assertTrue(deleted);
   }
 
+  @Test
+  void testReadJsonFileFromBucket() throws Exception {
+    String testJson = """
+        {
+          "institutionDomainMap": {
+            "Broad Institute": ["broadinstitute.org", "broad.mit.edu"],
+            "Harvard": ["harvard.edu", "hms.harvard.edu"]
+          }
+        }
+        """;
+
+    String blobIdName = "institution-domain-mapping.json";
+    Blob blob = mock(Blob.class);
+    when(blob.getContent()).thenReturn(testJson.getBytes());
+    when(storage.get(any(BlobId.class))).thenReturn(blob);
+
+    initStore();
+
+    InstitutionDomainMap result = service.readJsonFileFromBucket(blobIdName,
```

Comment:
This is probably OK but it does conflate testing the `InstitutionDomainMap`'s serialization with the GCS service testing. If the type changes, then this test will need to change. You could use a locally defined class instead if you wanted to isolate this test from `InstitutionDomainMap`